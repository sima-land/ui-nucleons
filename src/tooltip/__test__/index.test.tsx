import React from 'react';
import { render, fireEvent } from '@testing-library/react';
import { Tooltip } from '..';

describe('Tooltip', () => {
  it('should renders correctly', () => {
    const { container } = render(
      <Tooltip className='test-class' style={{ opacity: 0.2 }}>
        Hello, world!
      </Tooltip>,
    );

    expect(container).toMatchSnapshot();
  });

  it('should handle cross click', async () => {
    const spy = jest.fn();

    const { findByTestId } = render(<Tooltip onClose={spy}>Hello, world!</Tooltip>);

    expect(spy).toHaveBeenCalledTimes(0);

    fireEvent.click(await findByTestId('tooltip:cross'));

    expect(spy).toHaveBeenCalledTimes(1);
  });
});
