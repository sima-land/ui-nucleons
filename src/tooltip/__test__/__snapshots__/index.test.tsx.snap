// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tooltip should renders correctly 1`] = `
<div>
  <div
    class="root test-class"
    data-testid="tooltip"
    style="opacity: 0.2;"
  >
    <svg
      class="cross"
      data-testid="tooltip:cross"
      height="16"
      viewBox="0 0 16 16"
      width="16"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M12.707 4.707a1 1 0 0 0-1.414-1.414L8 6.586 4.707 3.293a1 1 0 0 0-1.414 1.414L6.586 8l-3.293 3.293a1 1 0 1 0 1.414 1.414L8 9.414l3.293 3.293a1 1 0 0 0 1.414-1.414L9.414 8l3.293-3.293Z"
      />
    </svg>
    <div
      class="content"
    >
      Hello, world!
    </div>
  </div>
</div>
`;
